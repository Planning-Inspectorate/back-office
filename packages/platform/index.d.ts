import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
