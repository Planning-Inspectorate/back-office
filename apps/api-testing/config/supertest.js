// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
