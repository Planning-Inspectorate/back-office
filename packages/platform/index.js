export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
