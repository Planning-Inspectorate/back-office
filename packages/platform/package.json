{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
