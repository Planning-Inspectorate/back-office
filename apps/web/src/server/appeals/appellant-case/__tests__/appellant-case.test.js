import { parseHtml } from '@pins/platform';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);
const baseUrl = '/appeals-service/appeal-details';
const pageUrl = '/appellant-case';

describe('appeal-details', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	describe('GET /appellant-case', () => {
		it('should render the appellant case page', async () => {
			const response = await request.get(`${baseUrl}/1${pageUrl}`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});
});
