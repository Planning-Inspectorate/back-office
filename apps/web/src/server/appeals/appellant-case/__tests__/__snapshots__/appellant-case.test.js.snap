// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appeal-details GET /appellant-case should render the appellant case page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-full\\"><span class=\\"govuk-caption-l govuk-!-margin-bottom-3\\">Appeal 351062</span>
            <h1             class=\\"govuk-heading-l govuk-!-margin-bottom-3\\">Appellant case</h1>
        </div>
    </div>
    <dl class=\\"govuk-summary-list govuk-summary-list--no-border\\">
        <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Site address</dt>
            <dd class=\\"govuk-summary-list__value\\">21 The Pavement, Wandsworth, SW4 0HY</dd>
        </div>
        <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Local planning authority</dt>
            <dd             class=\\"govuk-summary-list__value\\">Wiltshire Council</dd>
        </div>
    </dl>
    <div class=\\"govuk-summary-card\\">
        <div class=\\"govuk-summary-card__title-wrapper\\">
            <h2 class=\\"govuk-summary-card__title\\">1. The appellant</h2>
        </div>
        <div class=\\"govuk-summary-card__content\\">
            <dl class=\\"govuk-summary-list\\">
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Name</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>Eva Sharma</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Name</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Another individual</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>Eva Sharma</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Another individual</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Application reference</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>undefined</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Application reference</span></a>
                    </dd>
                </div>
            </dl>
        </div>
    </div>
    <div class=\\"govuk-summary-card\\">
        <div class=\\"govuk-summary-card__title-wrapper\\">
            <h2 class=\\"govuk-summary-card__title\\">2. The appeal site</h2>
        </div>
        <div class=\\"govuk-summary-card__content\\">
            <dl class=\\"govuk-summary-list\\">
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Site address</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>92 Huntsmoor Road, Tadley, RG26 4BX</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Site address</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Site fully owned</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>Yes</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Site fully owned</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Site partially owned</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>No</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Site partially owned</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> All owners known</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>No</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> All owners known</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Attempted to identify owners</dt>
                    <dd                     class=\\"govuk-summary-list__value\\"><span>No</span>
                        </dd>
                        <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Attempted to identify owners</span></a>
                        </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Advertised appeal</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>No</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Advertised appeal</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Visibility</dt>
                    <dd class=\\"govuk-summary-list__value\\"><span>Yes</span>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Visibility</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Site health and safety issues</dt>
                    <dd                     class=\\"govuk-summary-list__value\\"><span>No</span>
                        </dd>
                        <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Site health and safety issues</span></a>
                        </dd>
                </div>
            </dl>
        </div>
    </div>
    <div class=\\"govuk-summary-card\\">
        <div class=\\"govuk-summary-card__title-wrapper\\">
            <h2 class=\\"govuk-summary-card__title\\">3. The appeal</h2>
        </div>
        <div class=\\"govuk-summary-card__content\\">
            <dl class=\\"govuk-summary-list\\">
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Application form known</dt>
                    <dd class=\\"govuk-summary-list__value\\"><a href=\\"application-form.pdf\\" class=\\"govuk-link\\">application-form.pdf</a>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Application form known</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Decision letter</dt>
                    <dd class=\\"govuk-summary-list__value\\"><a href=\\"decision-letter.pdf\\" class=\\"govuk-link\\">decision-letter.pdf</a>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Decision letter</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Appeal statement</dt>
                    <dd class=\\"govuk-summary-list__value\\"><a href=\\"appeal-statement.pdf\\" class=\\"govuk-link\\">appeal-statement.pdf</a>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Appeal statement</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Add new supporting documents</dt>
                    <dd                     class=\\"govuk-summary-list__value\\"><span>Yes</span>
                        </dd>
                        <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> Add new supporting documents</span></a>
                        </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> New supporting documents</dt>
                    <dd                     class=\\"govuk-summary-list__value\\"><a href=\\"new-supporting-documents-1.pdf\\" class=\\"govuk-link\\">new-supporting-documents-1.pdf</a>
                        </dd>
                        <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> New supporting documents</span></a>
                        </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"></dt>
                    <dd class=\\"govuk-summary-list__value\\"><a href=\\"new-supporting-documents-2.pdf\\" class=\\"govuk-link\\">new-supporting-documents-2.pdf</a>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change</a>
                    </dd>
                </div>
            </dl>
        </div>
    </div>
    <div class=\\"govuk-form-group\\">
        <fieldset class=\\"govuk-fieldset\\">
            <legend class=\\"govuk-fieldset__legend govuk-fieldset__legend--m\\">What is the outcome of your review?</legend>
            <div class=\\"govuk-radios\\"
            data-module=\\"govuk-radios\\">
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"reviewOutcome\\" name=\\"reviewOutcome\\"
                    type=\\"radio\\" value=\\"valid\\">
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"reviewOutcome\\">Valid</label>
                </div>
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"reviewOutcome-2\\" name=\\"reviewOutcome\\"
                    type=\\"radio\\" value=\\"invalid\\">
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"reviewOutcome-2\\">Invalid</label>
                </div>
                <div class=\\"govuk-radios__item\\">
                    <input class=\\"govuk-radios__input\\" id=\\"reviewOutcome-3\\" name=\\"reviewOutcome\\"
                    type=\\"radio\\" value=\\"incomplete\\">
                    <label class=\\"govuk-label govuk-radios__label\\" for=\\"reviewOutcome-3\\">Incomplete</label>
                </div>
            </div>
        </fieldset>
    </div>
    <button type=\\"submit\\" class=\\"govuk-button\\" data-module=\\"govuk-button\\">Continue</button>
</main>"
`;
